sensors.py

```diff
@@ -20,28 +25,93 @@ def get_reading(self, sensor_type: SensorType) -> int | float:
         raise NotImplementedError
 
 class DHT(Sensor):
+    def __init__(self) -> None:
+        super().__init__()
+        self.dht = DHT22(board.D4)
+
     def get_reading(self, sensor_type: SensorType) -> int:
         with self._lock:
-            if sensor_type == SensorType.TEMPERATURE:
-                return randint(10, 30)
-            if sensor_type == SensorType.HUMIDITY:
-                return randint(40, 100)
+            try:
+                if sensor_type == SensorType.TEMPERATURE:
+                    return self.dht.temperature
+                if sensor_type == SensorType.HUMIDITY:
+                    return self.dht.humidity
+            except:
+                logging.exception("Couldn't get a reading")
+                raise SensorReadingError
         raise WrongSensorType
 
 class BMP280(Sensor):
+    def __init__(self) -> None:
+        super().__init__()
+        self.bmp280 = adafruit_bmp280.Adafruit_BMP280_I2C(board.I2C(), address=0x76)
+        
     def get_reading(self, sensor_type: SensorType) -> int | float:
         with self._lock:
-            if sensor_type == SensorType.PRESSURE:
-                return randint(990, 1100)
+            try:
+                if sensor_type == SensorType.PRESSURE:
+                    return int(self.bmp280.pressure)
+            except:
+                logging.exception("Couldn't get a reading")
+                raise SensorReadingError
         raise WrongSensorType
 
 class PMSA003C(Sensor):
+    def __init__(self) -> None:
+        super().__init__()
+        self.RX = 24
+        self.pi = pigpio.pi()
+        self.start1 = 0x42
+        self.start2 = 0x4d
+        self.working = True
+        self.pm = {SensorType.PM1: None, SensorType.PM2_5: None, SensorType.PM10: None}
+        
+        if not self.pi.connected:
+            logging.error("pigpio not connected!")
+            self.working = False
+            return
```

Comment:
Isn't that critical error? Maybe we should raise Exception here instead of return?
Please consider.

Reply:
I could but it would take more changes to make sure everything works even if one sensor doesn't.
Same with other sensors. Maybe open issue and if there is time I could make a change.
Right now if sensor fails during construction then application tries to read value anyway and fails(and just logs that to file, without adding that value to database).
I could change that so sensor is just never read.

Reply:
Sure. So I understand that if communication with one sensor fails it's not critical for the whole system right?
Maybe if it so, let's leave as is, if not add TODO?

Reply:
Up

Reply:
Maybe add issue, each sensor can fail and it will still work